08ca0dc9f174afdf8161f7f69310349e599c8a87
src/main/java/org/apache/platypus/server/grpc/LuceneServer.java
--- 
+++ 
@@ -342,7 +342,7 @@
                             pq.offer(gen);
                         }
                         long t1 = System.nanoTime();
-
+                        finishIndexingJob();
                         responseObserver.onNext(AddDocumentResponse.newBuilder().setGenId(String.valueOf(pq.peek())).build());
                         responseObserver.onCompleted();
                         logger.info(String.format("Indexing job completed for %s docs, in %s chunks, with latest sequence number: %s, took: %s micro seconds",
@@ -361,6 +361,12 @@
                         count = 0;
                     }
 
+                }
+
+                private void finishIndexingJob() throws IOException {
+                    for (String indexName : globalState.getIndexNames()) {
+                        globalState.getIndex(indexName).getShard(0).maybeRefreshBlocking();
+                    }
                 }
             };
         }